src/lib.rs

```diff
@@ -487,8 +471,7 @@ where
 {
     let mut receive_buf = [0u8; 1024];
     let mut b = backend_builder.build(continue_listening.clone(), &mut receive_buf[..]);
-    let mut flows = HashMap::<u32, U::Flow>::default();
-    let backend = b.sender();
+    let mut dp_to_flowmap = HashMap::<I::Addr, HashMap::<u32, U::Flow>>::new();
```

Comment:
If for some reason there was an `I::Addr` which could not `impl std::hash::Hash`, this might have to be split into two hashmaps. I can't think of a case where this might happen though, so I think we are good.

Reply:
Fair, but yeah can't imagine why that would be the case so hopefully we are fine at least for now